File: ToyLanguageInterpretor/src/model/statement/AssignStmt.java
package model.statement;
import model.Expression;
import model.ProgState;
import utils.*;
public class AssignStmt implements Statement{
	private Expression expr;
	private String var;
	
	public AssignStmt(Expression exp,String a){
		expr = exp;
		var = a;
	}
	@Override
	public ProgState execute(ProgState p){
		ISymbolTable <String,Integer> t = p.getSymbolTable();
		IHeap<Integer,Integer> hp = p.getHeap();
		int result = expr.evaluate(t,hp);
		t.add(var,result);
		return null ;
	}
	
	public String toString(){
		return " " + var + "=" + expr;
	}

}

File: C#/ToyLanguageC/model/AssignStmt.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ToyLanguageC.utils;

namespace ToyLanguageC.model
{
    class AssignStmt : Statement
    {
        private string name;
        private Expression exp;

        public AssignStmt(string name, Expression exp)
        {
            this.name = name;
            this.exp = exp;
        }
        public ProgState execute(ProgState state)
        {
            SymbolTable<string, int> table = state.getSymbolTable;
            int val = exp.evaluate(table);
            table.add(name, val);
            return state;
        }

        public string toString()
        {
            return name + "=" + exp.toString(); 
        }
    }
}
